src/hooks/Guilds/useVotes.ts

```diff
@@ -0,0 +1,88 @@
+import { useState, useEffect, useCallback } from 'react';
+import { useProposal } from 'hooks/Guilds/ether-swr/useProposal';
+import { BigNumber, bnum } from 'utils';
+import { useParams } from 'react-router-dom';
+import { useERC20Guild } from './contracts/useContract';
+import { useVotingPowerOf } from './ether-swr/useVotingPowerOf';
+import { useWeb3React } from '@web3-react/core';
+import { useTransactions } from 'contexts/Guilds';
+import { useGuildConfig } from './ether-swr/useGuildConfig';
+import { useERC20Info } from './ether-swr/erc20/useERC20Info';
+
+export interface VoteData {
+  args: unknown;
+  quorum: BigNumber;
+  totalLocked: BigNumber;
+  token: string;
+}
+
+interface useVotesReturns {
+  setVote: (action: BigNumber) => void;
+  voteData: VoteData;
+  flagCheckered: number;
+}
+
+export const useVotes = (): useVotesReturns => {
+  const [voteData, setVoteData] = useState<VoteData>({
+    args: {},
+    quorum: bnum(0),
+    totalLocked: bnum(0),
+    token: '',
+  });
+
+  const { guild_id: guildId, proposal_id: proposalId } =
+    useParams<{ guild_id?: string; proposal_id?: string }>();
+
+  const contract = useERC20Guild(guildId, true);
+  const { account } = useWeb3React();
+  const { createTransaction } = useTransactions();
+
+  // swr hooks
+  const { data: proposal } = useProposal(guildId, proposalId);
+  const {
+    data: { token, votingPowerForProposalExecution: quorum, totalLocked },

```

Comment:
We can't really de-structure code this way with SWR hooks as data is null while its being loaded. (If you refresh the single proposal page while its open, it crashes.)

Reply:
Good catch. I'll change this